share/completions/terraform.fish

```diff
@@ -15,60 +15,193 @@ function __fish_terraform_needs_command
     return 0
 end
 
+function __fish_terraform_using_command
```

Comment:
@faho should comment on this since I'm a completion newbie but I thought that functions in general should be using `__fish_use_subcommand` rather than rolling their own custom version. That there are existing completions that haven't been updated to use the global function is no surprise.